### PR TITLE
[glider-js] Refactor, update types

### PR DESCRIPTION
**Pull request by martin-badin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test glider-js`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/NickPiscitelli/Glider.js>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by martin-badin** [comment]
@peterblazejewicz Hi, please could you review?

File: types/glider-js/index.d.ts [hunk 1]
```diff
@@ -184,85 +186,97 @@ declare namespace Glider {
...
+    page: number;
+    preventClick: boolean;
+    slide: number;
+    slides: HTMLCollection;
+    track: HTMLElement;
+    trackWidth: number;
+
+    private _opt: Glider.Options;
+
+    constructor(ref: T, options: Glider.Options);
```
Comment (peterblazejewicz) [inline types/glider-js/index.d.ts:216]:
```suggestion
    constructor(ref: T, options?: Glider.Options);
```
```js
new Glider(document.querySelector('.glider'));
```
(from readme, works via `Object.assign`

**Comment by peterblazejewicz** [comment]
just 2c: still missing syntax support for this JS pattern (from readme):
```js
Glider(document.querySelector(element_path)).destroy();
```
(merge inerface with symbol to support new (...) and (...) functions)

**Comment by martin-badin** [comment]
> just 2c: still missing syntax support for this JS pattern (from readme):
> 
> ```js
> Glider(document.querySelector(element_path)).destroy();
> ```
> 
> (merge inerface with symbol to support new (...) and (...) functions)

Thanks. I pushed updates.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@martin-badin thx!

**Comment by martin-badin** [comment]
Ready to merge

**Pull request merged**